src/server/lib/okta/api/sessions.ts

```diff
@@ -22,31 +25,65 @@ const { okta } = getConfiguration();
  * @param sessionId Okta session ID
  * @returns Promise<SessionResponse>
  */
-export const getSession = async (
-	sessionId: string,
-): Promise<SessionResponse> => {
-	const path = buildUrl('/api/v1/sessions/:sessionId', { sessionId });
-	return fetch(joinUrl(okta.orgUrl, path), {
-		headers: { ...defaultHeaders, ...authorizationHeader() },
-	}).then(handleSessionResponse);
+export const getCurrentSession = async ({
+	sid,
+	idx,
+}: {
+	sid?: string;
+	idx?: string;
+}): Promise<SessionResponse> => {
+	const path = buildUrl('/api/v1/sessions/me');
+
+	const Cookie = `${sid ? `sid=${sid};` : ''}${idx ? `idx=${idx};` : ''}`;
+
+	const response = await fetch(joinUrl(okta.orgUrl, path), {
+		headers: { ...defaultHeaders, Cookie },
+		credentials: 'include',
+	});
+
+	if (response.ok) {
+		try {
+			return sessionSchema.parse(await response.json());
+		} catch (error) {
+			throw new OktaError({
+				message: 'Could not parse Okta session response',
```

Comment:
Would it be worth adding the response to the error message to help understand what went wrong?

Reply:
this error is only caught when the [zod](https://github.com/colinhacks/zod) [`.parse`](https://github.com/colinhacks/zod?tab=readme-ov-file#parse) method fails, where it just throws a generic error. Or if the [`response.json()`](https://developer.mozilla.org/en-US/docs/Web/API/Response/json_static#exceptions) throws an error, again if it fails to parse.

So we don't actually need to specifically handle the `error` variable itself.